R/summarize_weighted_production.R

```diff
@@ -187,10 +187,9 @@ summarize_unweighted_emission_factor <- function(data, ...) {
 }
 
 calculate_weighted_loan_metric <- function(data, metric) {
-  crucial <- c(metric, "loan_weight")
 
-  check_crucial_names(data, crucial)
-  walk_(crucial, ~ check_no_value_is_missing(data, .x))
+  check_crucial_names(data, c(metric, "loan_weight"))
+  walk_("loan_weight", ~ check_no_value_is_missing(data, .x))
```

Comment:
`calculate_weighted_loan_metric` is used on a number of metrics, not only `"production"`. Is it okay to remove `metric` in general from this expectation?

Reply:
Removing this expectation basically just means that and INPUT with `metric = NA` would give an output with `weighted_metric = NA`. Not necessarily ideal, but the other metrics calculated don't ever expect to have inputs with `NA`s. 

Is that OK, well I'm not sure. It shouldn't lead to any bugs if the input data is correct.
It may lead to a more difficult user experience if the input data is incorrect (e.g. one of the error messages will no longer be present).

Reply:
I guess one option would be to provide a boolean `check_metric` that allows toggling that check for the metric in each call. in that case we would not lose the check for all other metrics beside `"production"`, which seems the only one that should not use the check? also not a nicer interface, but it seems like a less drastic change in checks overall

Reply:
Done!